### DISCUSSION TITLE
Rewrite URL on the fly

### DISCUSSION THREAD

**Discussion by Zabrane** [mitmproxy/mitmproxy#6395] [General]
Hi guys,

I'm new to `mitmproxy` and I'd like to know if it is capable of handling these use-cases:

1.  `GET`
-  let's say I'd like to load a web page, be it: `https://www.wikipedia.org/`
- a `mitmproxy` hook will intercept that URL
- the URL is rewritten and sent to my local proxy

```
GET http://localhost:8000/https://www.wikipedia.org/ HTTP/1.1\r\n
[...] I don't touch the original HTTP request headers and keep them as they were
\r\n\r\n
```

2. `POST`
Same logic for POST requests. Intercept a POST query and rewrite it this way:

```
POST http://localhost:8000/https://reqbin.com/post-online/ HTTP/1.1\r\n
[...] I don't touch the original HTTP request headers and keep them as they were
\r\n\r\n
<POST BODY>
```

Is this doable in `mitmproxy`? If yes, can someone point me to the corresponding doc or any relevant code snippet?

Help appreciated
Thanks

**Comment by elespike** [comment]
Should be possible, try something like this:
https://docs.mitmproxy.org/stable/addons-examples/#http-redirect-requests

You probably also have to modify `request.path`, and you may need to URL-encode it.
